copy/copy.go

```diff
@@ -0,0 +1,165 @@
+package copy
```

Comment:
> Using a single image package would be very nice, but isn’t possible because of a dependency loop: copy depends on signature for creating new signatures, signature depends on transports for parsing Policy, transports depend on various transports, which depend on image.FromSource. I don’t know how to fix this cleanly. Perhaps we could hide image.FromSource into internalimageutils or something, have every external user go through ImageReference.NewImage instead of mostly calling image.FromSource as they do now, and then have image primarily for the copy code.

I would have loved this (fixing dep cycles is good in golang and interfaces) but I agree with what you wrote below, so nevermind..

> 
> OTOH, the copy implementation and image might need to change fairly radically for schema conversion etc. still, so I am not too eager to commit to a definitive package structure ATM. We can reshuffle this after the implementations are clearer in a few weeks.